src/main/java/com/openshift/internal/restclient/model/KubernetesResource.java

```diff
@@ -105,11 +105,11 @@ public void refresh(){
 		this.node = ModelNode.fromJSONString(client.get(getKind(), getName(), getNamespace()).toString());
 	}
 	
-	// TODO Pretty certain this should be protected
 	@Override
 	public ResourceKind getKind(){
-		if(node.has("kind")){
-			return ResourceKind.valueOf(node.get("kind").asString());
+		ModelNode kindNode = get(ResourcePropertyKeys.KIND);
+		if(kindNode.isDefined()){
+			return ResourceKind.valueOf(kindNode.asString());
```

Comment:
this will throw for unknown/new kinds that we are not aware of. We should either get away from enums or provide a lenient #valueOf